memcache/memcache.go

```diff
@@ -653,9 +657,12 @@ func (c *Client) GetMulti(keys []string, opts ...Option) (map[string]*Item, erro
 
 // parseGetResponse reads a GET response from r and calls cb for each
 // read and allocated Item
-func (c *Client) parseGetResponse(r *bufio.Reader, opts *Options, cb func(*Item)) error {
+func (c *Client) parseGetResponse(r *bufio.Reader, conn *conn, opts *Options, cb func(*Item)) error {
 	for {
+		// extend deadline before each additional call, otherwise all cumulative calls use the same overall deadline
+		conn.extendDeadline()
```

Comment:
Have you check what's the cost of extending the deadline? May it be impactful?

Reply:
AFAICS it's just modifying a timer [on the connection itself](https://github.com/grafana/gomemcache/blob/6947259a05865b6a7482cb73b556591498f3da0a/memcache/memcache.go#L242).

Under the hood I traced this down to [`setDeadlineImpl`](https://cs.opensource.google/go/go/+/master:src/internal/poll/fd_plan9.go;l=119?q=setdeadline&ss=go%2Fgo), looks fairly inexpensive to me.

Reply:
Ok, thanks for checking.